v2/api/machinelearningservices/v1api20240401/structure.txt

```diff
@@ -0,0 +1,8623 @@
+│   │   │   │   ├── VmPriority: *Enum (2 values)
+│   │   │   │   │   ├── "Dedicated"
+│   │   │   │   │   └── "LowPriority"
+│   │   │   │   └── VmSize: *string
+│   │   │   └── ResourceReference: *genruntime.ResourceReference
+│   │   ├── ComputeInstance: *Object (6 properties)
+│   │   │   ├── ComputeLocation: *string
+│   │   │   ├── ComputeType: *Enum (1 value)
+│   │   │   │   └── "ComputeInstance"
+│   │   │   ├── Description: *string
+│   │   │   ├── DisableLocalAuth: *bool
+│   │   │   ├── Properties: *Object (10 properties)
+│   │   │   │   ├── ApplicationSharingPolicy: *Enum (2 values)
+│   │   │   │   │   ├── "Personal"
+│   │   │   │   │   └── "Shared"
+│   │   │   │   ├── ComputeInstanceAuthorizationType: *Enum (1 value)
+│   │   │   │   │   └── "personal"
+│   │   │   │   ├── CustomServices: Object (6 properties)[]
+│   │   │   │   │   ├── Docker: *Object (1 property)
+│   │   │   │   │   │   └── Privileged: *bool
+│   │   │   │   │   ├── Endpoints: Object (5 properties)[]
+│   │   │   │   │   │   ├── HostIp: *string
+│   │   │   │   │   │   ├── Name: *string
+│   │   │   │   │   │   ├── Protocol: *Enum (3 values)
+│   │   │   │   │   │   │   ├── "http"
+│   │   │   │   │   │   │   ├── "tcp"
+│   │   │   │   │   │   │   └── "udp"
+│   │   │   │   │   │   ├── Published: *int
+│   │   │   │   │   │   └── Target: *int
+│   │   │   │   │   ├── EnvironmentVariables: map[string]Object (2 properties)
+│   │   │   │   │   │   ├── Type: *Enum (1 value)
+│   │   │   │   │   │   │   └── "local"
+│   │   │   │   │   │   └── Value: *string
+│   │   │   │   │   ├── Image: *Object (2 properties)
+│   │   │   │   │   │   ├── Reference: *string
+│   │   │   │   │   │   └── Type: *Enum (2 values)
+│   │   │   │   │   │       ├── "azureml"
+│   │   │   │   │   │       └── "docker"
+│   │   │   │   │   ├── Name: *string
+│   │   │   │   │   └── Volumes: Object (8 properties)[]
+│   │   │   │   │       ├── Bind: *Object (3 properties)
+│   │   │   │   │       │   ├── CreateHostPath: *bool
+│   │   │   │   │       │   ├── Propagation: *string
+│   │   │   │   │       │   └── Selinux: *string
+│   │   │   │   │       ├── Consistency: *string
+│   │   │   │   │       ├── ReadOnly: *bool
+│   │   │   │   │       ├── Source: *string
+│   │   │   │   │       ├── Target: *string
+│   │   │   │   │       ├── Tmpfs: *Object (1 property)
+│   │   │   │   │       │   └── Size: *int
+│   │   │   │   │       ├── Type: *Enum (4 values)
+│   │   │   │   │       │   ├── "bind"
+│   │   │   │   │       │   ├── "npipe"
+│   │   │   │   │       │   ├── "tmpfs"
+│   │   │   │   │       │   └── "volume"
+│   │   │   │   │       └── Volume: *Object (1 property)
+│   │   │   │   │           └── Nocopy: *bool
+│   │   │   │   ├── EnableNodePublicIp: *bool
+│   │   │   │   ├── PersonalComputeInstanceSettings: *Object (1 property)
+│   │   │   │   │   └── AssignedUser: *Object (2 properties)
+│   │   │   │   │       ├── ObjectId: *string
```

Comment:
Is this an Entra ObjectId? If so, may need to be configurable so that the user can be provisioned at the same time.

Reply:
Nice spotting! I've marked these and other fields to be imported.